ssl/statem/statem_clnt.c

```diff
@@ -1130,12 +1133,27 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             }
         }
 
+        /* Fix session version if it is not reused and we are not in EAP-FAST */
+        if (s->session->tlsext_tick == NULL)
```

Comment:
Hm... but version is still negotiated, even in EAP-FAST case, right? Couldn't it be that we got EAP master secret for higher version than what we just negotiated here? Or is it ok because it just sets `master_secret`, which is equally sized for all of TLS1.x?